assets/js/modules/ads/components/common/WooCommerceRedirectModal.test.js

```diff
@@ -369,4 +508,110 @@ describe( 'WooCommerceRedirectModal', () => {
 		).toBe( true );
 		expect( onDismiss ).toHaveBeenCalled();
 	} );
+
+	it( 'clicking "Continue with Google for WooCommerce" should trigger the correct internal tracking event when Google for WooCommerce is active with no Ads account linked', async () => {
+		fetchMock.postOnce( dismissItemEndpoint, {} );
+
+		const notification =
+			ADS_NOTIFICATIONS[ 'account-linked-via-google-for-woocommerce' ];
+
+		registry
+			.dispatch( CORE_NOTIFICATIONS )
+			.registerNotification(
+				'account-linked-via-google-for-woocommerce',
+				notification
+			);
+
+		registry.dispatch( MODULES_ADS ).receiveModuleData( {
+			plugins: {
+				[ PLUGINS.WOOCOMMERCE ]: {
+					active: true,
+					installed: true,
+				},
+				[ PLUGINS.GOOGLE_FOR_WOOCOMMERCE ]: {
+					active: true,
+					installed: true,
+					adsConnected: false,
+				},
+			},
+		} );
+		const { container, waitForRegistry, getByText } = render(
+			<ModalComponent />,
+			{
+				registry,
+				viewContext: VIEW_CONTEXT_MAIN_DASHBOARD,
+			}
+		);
+
+		await waitForRegistry();
+
+		expect(
+			container.querySelector(
+				'.mdc-button:not(.mdc-dialog__cancel-button)'
+			)
+		).toHaveAttribute(
+			'href',
+			'http://example.com/wp-admin/admin.php?page=wc-admin&path=%2Fgoogle%2Fdashboard'
+		);
+
+		const useGoogleForWooCommerceButton = getByText(
+			/Use Google for WooCommerce/i
+		);
+
+		fireEvent.click( useGoogleForWooCommerceButton );
+
+		await waitForRegistry();
+
+		expect( mockTrackEvent ).toHaveBeenCalledWith(
+			`${ VIEW_CONTEXT_MAIN_DASHBOARD }_pax_wc-redirect`,
+			'choose_gfw',
+			'gfw'
+		);
+	} );
+
+	it( 'clicking "Continue with Google for WooCommerce" should trigger the correct internal tracking event when Google for WooCommerce is active with Ads account linked', async () => {
+		fetchMock.postOnce( dismissItemEndpoint, {} );
+
+		const notification =
+			ADS_NOTIFICATIONS[ 'account-linked-via-google-for-woocommerce' ];
+
+		registry
+			.dispatch( CORE_NOTIFICATIONS )
+			.registerNotification(
+				'account-linked-via-google-for-woocommerce',
+				notification
+			);
+
+		registry.dispatch( MODULES_ADS ).receiveModuleData( {
+			plugins: {
+				[ PLUGINS.WOOCOMMERCE ]: {
+					active: true,
+					installed: true,
+				},
+				[ PLUGINS.GOOGLE_FOR_WOOCOMMERCE ]: {
+					active: true,
+					installed: true,
+					adsConnected: true,
+				},
+			},
+		} );
+		const { waitForRegistry, getByText } = render( <ModalComponent />, {
+			registry,
+			viewContext: VIEW_CONTEXT_MAIN_DASHBOARD,
+		} );
+
+		await waitForRegistry();
+
+		mockTrackEvent.mockClear();
+
+		const viewCurrentAdsAccountButton = getByText(
+			/View current Ads account/i
+		);
+
+		fireEvent.click( viewCurrentAdsAccountButton );
+
+		await waitForRegistry();
+
+		expect( mockTrackEvent ).not.toHaveBeenCalled();
```

Comment:
This should also track `choose_gfw`. In all scenarios, we should be tracking one of these two according to what the user chooses in the modal.

Reply:
Fixed in my latest iteration, thanks!